instance/models/database_server.py

```diff
@@ -47,10 +47,44 @@
 
 # Models ######################################################################
 
+
 class DatabaseServerManager(SharedServerManager):
     """
     Custom manager for the DatabaseServer model.
     """
+    def create_database_server(self, host=None, user=None, password=None, port=None,
+                               primary=None, replica_set=None):
+        """
+        Create MongoDB server
+        """
+        logger.info("Creating DatabaseServer %s", host)
+        extra_args = {}
+        if self.model._meta.model_name == 'mongodbserver':
```

Comment:
@rocioar This is essentially a type check, which is in general discouraged in object-oriented programming.

Maybe it should be a method with overrides in descendent classes? Something like

```
# this class
def extra_args(self, arg1, arg2, arg3, ...):
    return {}

# here
extra_args = self.extra_args(**necessary_parameters

# in MongoDBServerManager (create if not exists)
def extra_args(self, host, primary, replica_set):
    return dict(...)
```

Reply:
fixed :)